wavedump/src/fit-histograms

```diff
@@ -0,0 +1,566 @@
+    plt.legend()
+    plt.show()
+    input()
+    exit()
+
+if __name__ == '__main__':
+    from argparse import ArgumentParser
+    import matplotlib.pyplot as plt
+
+    parser = ArgumentParser(description='Fit SPE and sodium charge histograms')
+    parser.add_argument('filenames',nargs='+',help='input filenames (ROOT format)')
+    parser.add_argument('--sodium', default=False, action='store_true', help='flag to indicate data is from a sodium source')
+    parser.add_argument('--plot', default=False, action='store_true', help='plot the waveforms and charge integral')
+    parser.add_argument('--BV', type=int,default=42, help='Bias Voltage input')
+    #parser.add_argument('--peakper', type=double, default=0, help='Direct change of maximum percentage change from maximum peak')
+    parser.add_argument('--pdf', default=False, action='store_true', help='Save the fit canvas as pdf')
+    parser.add_argument('--ObjKeep', type=int,default=2,  help='Number of objects to fit')
+    # parser.add_argument('--SPE_FLL', default=-0.1, type=float, help='Left limit on the spe fit')
+    # parser.add_argument('--SPE_FLR', default=-0.1, type=float, help='Right limit on the spe fit')
+    parser.add_argument('--NA_GF', default=50, type=float, help='Right limit on the spe fit')
+    parser.add_argument('--chi2', default=False, action='store_true', help='Save the ndof and chi2 value of the fit to csv')
+    parser.add_argument('--root_func', default=False, action='store_true')
+
+    args = parser.parse_args()
+    
+    if not args.plot:
+        # Disables the canvas from ever popping up
+        gROOT.SetBatch()
+    
+    canvas = []
+    canvas_count = 1
+    if os.path.exists('temp_data.csv'):
+        os.remove('temp_data.csv')
+
+    for filename in args.filenames:
+        f = ROOT.TFile(filename,"UPDATE")
+        keycount=0
+        
+        # raw is unfiltered
+        raw_histograms = {}
+        filtered_histograms = {}
+        for key in f.GetListOfKeys():
+            #print("***KEYPRINT")
+            #print(key)
+            #print(key.GetTitle())
+            if (keycount>0):
+                if(key.GetTitle()==keyold.GetTitle()):
+                   # print("EQUAL")
+                    continue
+            keycount+=1
+            keyold=key
+            #if keycount>2:
+            #    break
+
+            #if keycount>args.ObjKeep:# Only gets channel 1
+            #    break
+                # Argument to delete all previous fits. Not working, related to TF1 class?
+                # Each time the code executes 2 new keys are created in the file.
+                #h.Reset("ICESM")
+
+            h = key.ReadObj()
+
+            if h.ClassName() != 'TH1F' and h.ClassName() != 'TH1D':
+                continue
+
+            if h.GetName().startswith('f_'):
+                filtered_histograms[h.GetName()] = h
+                keycount -= 1
+            else:
+                raw_histograms[h.GetName()] = (h, keycount)
```

Comment:
I'm a little confused about all the code above. What is the keycount doing? and what is the general purpose of checking class names?

Reply:
I did some experimenting, and it seems that every time we preform a fit, a new object is written to the root file that contains the drawing of the fit. When we run `fit-histograms`, we want to preform a brand-new fit, and so `keycount` and `keyold` ensure that we're fitting the first histograms, those being the ones that don't have any functions drawn on them.

That was the old way of doing it, which I did not write. I modified this to be much less complicated.

Also, the class check is to ensure that we're only reading histograms from the file. From my experimenting, it doesn't seem like this check ever catches anything. I've kept it because it was there before I started in June.